### PR TITLE
Naming improvements in Oracle

### PR DESCRIPTION
**Pull request by maurelian**
**Description**

The main changes are: 

1. Rename `sequencer` to `proposer`
3. Use `propose` instead of `append`
2. Use `outputRoot` instead of `l2Output` when referring to the `bytes32` hash of the Output

### CODE REVIEW

File: packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol [hunk 1]
```diff
@@ -103,10 +103,10 @@ contract L2OutputOracle is OwnableUpgradeable, Semver {
     mapping(uint256 => OutputProposal) internal l2Outputs;
 
     /**
-     * @notice Reverts if called by any account other than the sequencer.
```
Comment (tynes) [inline packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol:106]: Perhaps we want to rename the `l2Outputs` mapping as well?

**Comment by tynes** [comment]
What do you think about also renaming `getL2Output` to be more in style?

**Comment by tynes** [comment]
I like the idea here

**Comment by tynes** [comment]
Looks like a new gas snapshot is necessary

File: packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol (hunk 1 above)
Comment (maurelian) [reply packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol:106]: I did play around with that, but didn't find anything I liked more. I considered: 
- `outputProposals`: but they are only proposals until they are finalized then they are outputs
- `outputs`: I do find the `l2` prefix a bit clunky and repetitive in this contract. However `output` is a vague term, and `l2` makes it more readable especially for external devs.

**Comment by maurelian** [comment]
> What do you think about also renaming `getL2Output` to be more in style?

What name do you have in mind? 
FWIW I did try `getOutput` as well as `proposeOutput`, `deleteOutput` etc. See my [comment here](https://github.com/ethereum-optimism/optimism/pull/2990#discussion_r919060561) about removing `l2`.

**Review by smartcontracts — APPROVED** [review]
Nice, good changes.

**Comment by tynes** [comment]
@mergifyio refresh

**Comment by maurelian** [comment]
> @mergifyio refresh

What does that do @tynes ?

**Review by protolambda — APPROVED** [review]
LGTM. You may need to update the bindings or gas-snapshot later, depending on the merge order with the other PRs in review currently.

**Review by tynes — APPROVED** [review]

**Pull request merged**